src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -100,16 +101,17 @@
         job.getTasks()
            .forEach(task -> task.getVariables()
                                 .values()
-                                .forEach(taskVariable -> addDependencyIfCatalogObjectModelExist(keyValueMapBuilder,
-                                                                                                taskVariable)));
+                                .forEach(taskVariable -> addDependsOnIfCatalogObjectModelExistOnTaskVariable(keyValueMapBuilder,
+                                                                                                             taskVariable)));
+
         addJobDescriptionIfNotNullAndNotEmpty(keyValueMapBuilder, job);
         addJobVizualisationIfNotNullAndNotEmpty(keyValueMapBuilder, job);
 
-        return keyValueMapBuilder.build();
+        return new ArrayList<>(keyValueMapBuilder.build());
     }
 
     private void addProjectNameIfNotNullAndNotEmpty(
-            ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, Job job) {
+            ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, Job job) {
```

Comment:
Fabien noticed in general in case of using Set, the order is not guaranteed. I'm not sure if the order of metadata is used somewhere, but before it was ordered in List. 
We can use LinkedHashSet, but it seems that Guava ImmutableSet already provides order according to this answer 
https://stackoverflow.com/questions/25005595/class-that-describes-a-immutable-ordered-set
https://github.com/google/guava/wiki/ImmutableCollectionsExplained
and my tests. 
So we can leave like this.